### DISCUSSION TITLE
Welcome to facebook-scraper Discussions!

### DISCUSSION THREAD

**Discussion by kevinzg** [kevinzg/facebook-scraper#145] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by 228282** [comment]
Hello, dear friend. I have a question. There is a post where many pictures are attached, but with the help of your script I can only take the link first. How can you get links of all photos?

The script does not collect all the photos from the post. There is a post with a lot of photos, and one of them has a number with a plus (+7, for example). The script will collect links of those photos that are displayed, and the rest are not

**Reply by kevinzg** [reply]
Someone implemented the extraction of various photos per post a while ago, I guess he didn't notice that case.
You could create an issue and give an example of a post with many pictures attached.

**Comment by paulofaia2** [comment]
Hi! 
Congratulations!
Amazing job! :)

In next release, pleaseeeee, implement comments! :)
i need this!!! i can pay for you do it fast?

let's talk? call me: eu@paulofaia.com

**Reply by kevinzg** [reply]
Hi, I replied to your mail.

**Reply by neon-ninja** [reply]
See https://github.com/kevinzg/facebook-scraper/pull/188

**Comment by DesignMike** [comment]
Can I run this on Google Collab? I'm getting some errors. https://prnt.sc/x5l639

**Reply by kevinzg** [reply]
Haven't used that product, but looks like a Jupyter notebook, so you should be writing Python code.
The code you pasted is meant to be used from a command line shell like Bash.

**Comment by DesignMike** [comment]
Can you extract usernames, profile names, Email, gender, location, birthday, Phone?

**Reply by kevinzg** [reply]
Nope

**Reply by neon-ninja** [reply]
As of https://github.com/kevinzg/facebook-scraper/pull/201, now you can

**Comment by quant-ru** [comment]
Hello, @kevinzg. Some of facebook pages now doesnt have the ```/posts``` link. How to fix this?

**Reply by neon-ninja** [reply]
Should be fine as of https://github.com/kevinzg/facebook-scraper/pull/182

**Comment by evildrome** [comment]
+1 on the need for comments downloading.  :)

Thanks!

**Reply by neon-ninja** [reply]
As of https://github.com/kevinzg/facebook-scraper/pull/188, now you can

**Comment by ong-andrew** [comment]
I use your code to alert my team of new facebook posts by our competitors. 

Thank you for your work.

**Comment by SGeorgilas** [comment]
I have a question.I am requesting 800 pages per user because i need around 3200 posts of each one. But after i take posts from one user , i receive exception of 443 from facebook...how can i solve this issue?

**Reply by neon-ninja** [reply]
This PR (https://github.com/kevinzg/facebook-scraper/pull/212) might be useful for that kind of scale

**Reply by SGeorgilas** [reply]
Thank you for your quick answer. It doesn't help me that much cause i have no problem with the time that i am fetching the posts. I have a problem of playing around the cooldown so i dont take a timeout.

**Reply by neon-ninja** [reply]
My thinking is that by making fewer requests, requesting larger amounts of posts per request, you're less likely to trigger request number based temporary IP bans

**Reply by neon-ninja** [reply]
Instead of making 800 requests of 4 posts per request, you could make 16 requests of 200 posts per request

**Reply by SGeorgilas** [reply]
Ok i get what you are thinking.I upgraded the package to 0.2.29, and my code looks like this:

get_posts = facebook_scraper.get_posts(fb_user,pages=16,timeout=10,options={"allow_extra_requests": False, "posts_per_page":200})

but it only fetches 63-64 posts.So what is wrong?

**Reply by neon-ninja** [reply]
https://github.com/kevinzg/facebook-scraper/pull/212 isn't merged yet, so it isn't in 0.2.29

**Reply by neon-ninja** [reply]
You should be able to install this PR with the command `pip install git+https://github.com/kevinzg/facebook-scraper.git@refs/pull/212/merge`.

**Reply by SGeorgilas** [reply]
Did it and still does not work.

**Reply by neon-ninja** [reply]
You might need to restart your Python kernel after updating the library

**Reply by neon-ninja** [reply]
Also make sure that the page or profile doesn't legitimately only have 64 posts
